### PR TITLE
Optional Icons added to DialogModal title

### PR DESCRIPTION
**Pull request by nataliepina**
Closes D2IQ-80933

Icon will be inserted to the left of the title, you can customize the icon size, color, and type. 

Note: I am not updating the story here to Controls, as there will be an update coming out soon on Storybook 6.4 that reforms the way to write CSF, will include these updates in the ui-kit chores ticket for updating the stories and controls. This is a rather large story that will have some more complex Controls to add and would be best to be separated out anyway. 

## Screenshots

Screen Shot 2021-11-10 at 10 23 41 AM

## Checklist for PR creator

- [ ] If any new components were added, there are exported from `packages/index.ts`
- [x] If this PR is associated with a JIRA, it is mentioned in commit message footer ("Closes …")
- [ ] If this PR contains breaking changes, is stated in commit message body ("BREAKING CHANGE: …")
- [x] Info for applicable sections above is provided

### CODE REVIEW

File: packages/modal/components/DialogModal.tsx [hunk 1]
```diff
@@ -20,6 +24,12 @@ export interface DialogModalProps extends ModalBaseProps {
   isContentFlush?: boolean;
   /** The text displayed in the header of the modal. */
   title: React.ReactNode;
+  /** Controls the icon type displayed next to the title. */
+  iconShape?: IconShapes | React.ReactElement<HTMLElement>;
+  /** Icon color can be customized otherwise it will inherit color. */
+  iconColor?: React.CSSProperties["color"];
+  /** Icon size can be customized, it is defaulted to "xs" */
+  iconSize?: IconSize;
```
Comment (mstruebing) [inline packages/modal/components/DialogModal.tsx:32]: 💬 What do you think about converting this to an object like:

icon: {
  shape: ...;
  color: ...;
  size: ...;
}
I'm not to opinionated about this, but it instantly came to my mind after seeing the repeated icon prefix.

Comment (vacas5) [reply packages/modal/components/DialogModal.tsx:32]: I like Max's idea too. Makes it more extensible.

File: packages/modal/components/DialogModal.tsx [hunk 2]
```diff
@@ -38,8 +51,28 @@ class DialogModal extends React.PureComponent<DialogModalProps, {}> {
...
+                    <IconPropAdapter
+                      icon={iconShape}
+                      size={iconSize}
+                      color={iconColor}
+                    />
+                  </FlexItem>
+                  <FlexItem flex="grow">{title}</FlexItem>
+                </Flex>
+              ) : (
+                <>{title}</>
```
Comment (vacas5) [inline packages/modal/components/DialogModal.tsx:73]: does this not work without the `<>` ?

Comment (nataliepina) [reply packages/modal/components/DialogModal.tsx:73]: Yeah, TypeScript 😓 we have a lot of this going on in Kommander too.

**Review by mstruebing — APPROVED** [review]
🎉

File: packages/modal/stories/helpers/ModalStoryContainer.tsx [hunk 3]
```diff
@@ -5,50 +5,21 @@ export interface ModalStoryContainerProps {
   children: any;
 }
 
-export interface ModalStoryContainerState {
```
Comment (clintonmedbery) [inline packages/modal/stories/helpers/ModalStoryContainer.tsx:8]: Yeeeessssss no more lifecycle components!

**Review by clintonmedbery — APPROVED** [review]
Nice!

**Review by clintonmedbery — APPROVED** [review]

**Pull request merged**